### PR TITLE
updated react version and type

### PR DESCRIPTION
**Pull request by arminakvn**
In order to update our henshu-app library, we need to upgrade the react and react-dom, etc. in the the henshu library itself and make publish to npm.

### CODE REVIEW

File: dist/elements/Each.d.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { ReactElement } from 'react';
 import { HenshuElementProps } from '../henshu';
 import { GetterSetter } from '../context';
-export declare type HenshuEachProps = HenshuElementProps & {
+export type HenshuEachProps = HenshuElementProps & {
     children(bindTo: (key: string) => GetterSetter, i: number): ReactElement;
     max?: number;
 };
-export default function Each(props: HenshuEachProps): JSX.Element;
+export default function Each(props: HenshuEachProps): import("react/jsx-runtime").JSX.Element;
```
Comment (nickcaler) [inline dist/elements/Each.d.ts:8]: Is there a way to write this so we aren't doing import inline with the export?  If not don't worry, but it seems hacky to import the type just as we export the function.

Comment (nickcaler) [reply dist/elements/Each.d.ts:8]: Same for the other elements.

File: src/elements/Each.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ export default function Each(props: HenshuEachProps) {
     const [items, setItems] = useState(get());
     const [selection, setSelection] = useState<Selection|null>(null);
     const [selectionBox, setSelectionBox] = useState<Element|null>(null);
-    const timeout = useRef();
+    const timeout = useRef<number | undefined>();
```
Comment (nickcaler) [inline src/elements/Each.tsx:26]: @arminakvn is this the only real change in this PR?   Are the changes in dist just the compiled version?  If so I think we may want to git ignore the dist folder from source control so the changes on generated code are not stored.

**Review by nickcaler — APPROVED** [review]
This looks good, just a couple questions on how we are importing, and on what we push to git.

File: src/elements/Each.tsx (hunk 2 above)
Comment (arminakvn) [reply src/elements/Each.tsx:26]: Thank you so much for your comments! Yes, that is the only code change-- the upgraded packages in the package.json file are what is needed for updating the henshu-app package (which is going to be coming after this goes through) -- regarding the ignoring the dist folder, that is actually something I've seen we do all different kinds of ways and I am not sure what is best practice? I usually just keep it the way I clone down the repositories, and in this case the dist is not ignored and I thought maybe that's a practice for npm libraries or something (in fact I think most of the packages in lib folder are like this). What is your suggestion going forward?

**Pull request merged**